### PR TITLE
Search should perform simple substring matching

### PR DESCRIPTION
**Pull request by Andrewjeska**

### CODE REVIEW

**Comment by xplato** [comment]
@Andrewjeska looks like CI is mad at you 😅

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx [hunk 1]
```diff
@@ -60,10 +60,9 @@ const defaultAccessCodeFilter = (
 ): boolean => {
   const value = searchInputValue.trim()
```
Comment (razor-x) [inline src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:61]:
```suggestion
  const value = searchInputValue.trim().toLowerCase()
```

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx [hunk 2]
```diff
@@ -60,10 +60,9 @@ const defaultAccessCodeFilter = (
...
   if (value === '') return true
-  const searchRegex = new RegExp(value, 'i')
   return (
-    searchRegex.test(accessCode?.name ?? '') ||
-    searchRegex.test(accessCode?.code ?? '')
+    (accessCode.name?.includes(value) || accessCode.code?.includes(value)) ??
+    false
```
Comment (razor-x) [inline src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:65]: How about something like

```ts
const { name = '', code = '' } = accessCode
return name.trim().toLowerCase().includes(value) || code.trim().toLowerCase().includes(value)
```

Comment (Andrewjeska) [reply src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx]: [028adf9](https://github.com/seamapi/react/pull/351/commits/028adf9a9e9da4e7fc9eecd6a7bdca68b1ba0a1a)

**Comment by Andrewjeska** [comment]
Screenshot 2023-08-16 at 1 27 23 PM
oops

File: src/lib/seam/components/DeviceTable/DeviceTable.tsx [hunk 3]
```diff
@@ -48,7 +48,7 @@ const defaultDeviceFilter = (
 ): boolean => {
   const value = searchInputValue.trim()
   if (value === '') return true
-  return new RegExp(value, 'i').test(device.properties.name ?? '')
+  return device.properties.name.includes(value)
```
Comment (razor-x) [inline src/lib/seam/components/DeviceTable/DeviceTable.tsx:51]:
```suggestion
  return device.properties.name.toLowerCase().includes(value)
```

File: src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx [hunk 4]
```diff
@@ -58,12 +58,12 @@ const defaultAccessCodeFilter = (
...
+  const value = searchInputValue.trim().toLowerCase()
   if (value === '') return true
-  const searchRegex = new RegExp(value, 'i')
+  const { name = '', code = '' } = accessCode
   return (
-    searchRegex.test(accessCode?.name ?? '') ||
-    searchRegex.test(accessCode?.code ?? '')
+    name.trim().toLowerCase().includes(value) ??
+    code?.trim().toLowerCase().includes(value)
   )
```
Comment (razor-x) [inline src/lib/seam/components/AccessCodeTable/AccessCodeTable.tsx:67]: My fault for forgetting about null vs undefined default. 

```ts
const name = accessCode.name ?? ''
const code = accessCode.code ?? ''
  return (
    name.trim().toLowerCase().includes(value) ||
    code.trim().toLowerCase().includes(value)
  )
  )

**Review by razor-x — APPROVED** [review]

**Comment by razor-x** [comment]
That's okay I think it should be safe to merge this one.

**Pull request merged**